python/ray/internal/internal_api.py

```diff
@@ -11,6 +11,24 @@ def global_gc():
     worker.core_worker.global_gc()
 
 
+def memory():
+    """Returns a formatted string describing memory usage in the cluster."""
+
+    import grpc
+    from ray.core.generated import node_manager_pb2
+    from ray.core.generated import node_manager_pb2_grpc
+
+    # We can ask any Raylet for the global memory info.
+    raylet = ray.nodes()[0]
+    raylet_address = "{}:{}".format(raylet["NodeManagerAddress"],
+                                    ray.nodes()[0]["NodeManagerPort"])
+    channel = grpc.insecure_channel(raylet_address)
+    stub = node_manager_pb2_grpc.NodeManagerServiceStub(channel)
+    reply = stub.FormatGlobalMemoryInfo(
+        node_manager_pb2.FormatGlobalMemoryInfoRequest(), timeout=30.0)
+    return reply.memory_summary
```

Comment:
Hmm sorry this is coming a bit late, but I wonder if we should be returning a data structure instead of a formatted string. It's much nicer to be able to programmatically work with the result, e.g., for dumping to the dashboard or plotting.

Reply:
Never mind, just saw the note about using `GetNodeStats` directly.